NioUIKit/Menu/MenuAccountPickerView.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  MenuAccountPickerView.swift
+//  Nio
+//
+//  Created by Finn Behrens on 30.03.22.
+//
+
+import MatrixCore
+import NioKit
+import SwiftUI
+
+public struct MenuAccountPickerContainerView: View {
+    @Binding var currentAccount: String
+
+    @AppStorage("showAccountsInPicker") var showAccounts: Bool = false
+
+    public var body: some View {
+        VStack(alignment: .leading) {
+            Button(action: {
+                showAccounts.toggle()
+            }) {
+                Image(systemName: "greaterthan")
+                    .foregroundColor(.gray)
+                    .imageScale(.medium)
+                    .rotationEffect(showAccounts ? .degrees(90) : .zero)
+            }.padding(.bottom)
+
+            if showAccounts {
+                MenuAccountPickerView(currentAccount: $currentAccount)
+                    .padding(.leading)
+            }
+        }
+    }
+}
+
+public struct MenuAccountPickerView: View {
+    @Binding var currentAccount: String
+
+    @FetchRequest(sortDescriptors: []) var accounts: FetchedResults<MatrixAccount>
+
+    init(currentAccount: Binding<String>) {
+        _currentAccount = currentAccount
+
+        _accounts = FetchRequest<MatrixAccount>(sortDescriptors: [NSSortDescriptor(key: "userID", ascending: true)], predicate: NSPredicate(format: "userID != %@", currentAccount.wrappedValue))
+    }
+
+    public var body: some View {
+        VStack(alignment: .leading) {
+            ForEach(accounts, id: \.userID) { account in
+                Button(action: {
+                    print("switching account to \(account.userID ?? "Unknown user")")
+                    currentAccount = account.userID ?? ""
+                }) {
+                    MenuAccountPickerAccountView(account: account).tag(account.userID ?? "Unknown user")
+                }.padding(.vertical)
```

Comment:
This would be my personal style preference for this, but feel free to ignore 🙃
```suggestion
                Button {
                    print("switching account to \(account.userID ?? "Unknown user")")
                    currentAccount = account.userID ?? ""
                } label: {
                    MenuAccountPickerAccountView(account: account).tag(account.userID ?? "Unknown user")
                }
                .padding(.vertical)
```